### ISSUE TITLE
Laravel Nova

### ISSUE THREAD

**Issue by johannesschobel** [qcod/laravel-app-settings#5]
Hey @saqueib ,

that is quite a nice package - great work.. I have a suggestion, though.. Wouldn't it be better to "remove" the UI part in a separate package? This would, in turn, provide the functionality to add a `laravel-nova` package for creating the UI as well.

Because, actually, `Laravel Nova` provides most of the UI Builder functionality you have already created..

All the best and cheers

**Comment by saqueib** [comment]
thanks @johannesschobel,

Actually I created this packages since there is already some package which has DB persistence option. But all are lacking the UI which is a must for most of the app which I create. 

Sure you can use it without the UI https://github.com/qcod/laravel-app-settings#use-without-ui

Laravel Nova is tottally different concept and its paid 🤑 , its admin panel with Resource management and tons of other features. This one is for anybody who just needs a settings management system with UI.

**Comment by saqueib** [comment]
You know what, its good suggestion. Extracting storage layer to another package should not hurt. I will look into this and will do it in next version. 👍

**Comment by johannesschobel** [comment]
yeah, i know that it is paid.. however, i thought about splitting up the package into:
1) `laravel-app-settings` -> the "storage engine"
2) `laravel-app-settings-ui`-> the custom ui builder you have that consumes the `laravel-app-settings` storage engine
3) `laravel-app-settings-nova` -> a nova package that consumes the `laravel-app-settings` package

All the best

**Comment by akiyamaSM** [comment]
@johannesschobel @saqueib 
There is already a Laravel Nova version I made, has been released one day later
https://github.com/akiyamaSM/laravel-nova-configuration
its easy to use too.

**Comment by johannesschobel** [comment]
yeah, but for future use it may be good to "split up" the package into a "ui" and "non-ui" version.. Because in API projects (with no direct UI) you will not need the UI Builder ;)

**Comment by johannesschobel** [comment]
@akiyamaSM the package you have mentioned does not use the `laravel-app-setting` package provided by this repository, right?! It is a complete stand-alone package, right?

**Comment by saqueib** [comment]
Sure, I will split this into storage and ui packages

**Comment by akiyamaSM** [comment]
> @akiyamaSM the package you have mentioned does not use the `laravel-app-setting` package provided by this repository, right?! It is a complete stand-alone package, right?

Exactly, Its a stand alone app

**Comment by saqueib** [comment]
@johannesschobel As you suggestions now this package has been split into two parts.

`qcod/laravel-app-settings` to keep the current API without any breaking changes and
`qcod/laravel-settings` for a key-value pair in the database

**Comment by johannesschobel** [comment]
Thank you.. that is awesome news
